src/routes/walletRoutes.js

```diff
@@ -24,25 +24,28 @@ class WalletRoutes extends BaseRoute {
             },
             handler: async (request) => {
                 try {
-                    const { id } = request.query;
-                    
-                    const result = await this.db.read(id);
+                    const {
+                        id
+                    } = request.params;
+
+                    const result = await this.db.read({id: id});
 
                     if (result.n <= 0) {
                         return Boom.preconditionFailed('ID não encontrado no banco')
                     }
                     return result;
-                }
-                catch (error){
-                    console.error('Erro na requisição da carteira!');
+                } catch (error) {
+                    console.error('Erro na requisição da carteira!', error);
                     return Boom.internal();
                 }
             }
 
         }
     };
 
-    purchase() {
+    // purchase() {
```

Comment:
Se você não for usar um código melhor só apagar mesmo. Evita só comentar que acaba com muito lixo em projetos longos